app/helpers/maintenance_tasks/tasks_helper.rb

```diff
@@ -102,29 +102,40 @@ def csv_file_download_path(run)
     end
 
     # Resolves values covered by the inclusion validator for a task attribute.
-    # Only Arrays are supported, option types such as:
-    # Procs, lambdas, symbols, and Range are not supported and return nil.
+    # Supported option types:
+    # - Arrays
+    # - Procs and lambdas that take no arguments and return an Array
+    # - Methods that take no arguments and return an Array. The method is called on the task instance.
+    # Other types are not supported and will return nil.
     #
     # Returned values are used to populate a dropdown list of options.
     #
     # @param task_class [Class<Task>] The task class for which the value needs to be resolved.
     # @param parameter_name [String] The parameter name.
     #
     # @return [Array] value of the resolved inclusion option.
-    def resolve_inclusion_value(task_class, parameter_name)
+    def resolve_inclusion_value(task, parameter_name)
+      task_class = task.class
       inclusion_validator = task_class.validators_on(parameter_name).find do |validator|
         validator.kind == :inclusion
       end
       return unless inclusion_validator
 
       in_option = inclusion_validator.options[:in] || inclusion_validator.options[:within]
+      in_option = in_option.call if in_option.is_a?(Proc) && in_option.arity.zero?
+
+      if in_option.is_a?(Symbol) && task.respond_to?(in_option)
+        method = task.method(in_option)
+        in_option = method.call if method.arity.zero?
```

Comment:
Let's just support all the normal resolution for this: https://github.com/rails/rails/blob/8aee12ec51728a6d8472335ec9d4ad68d0521773/activemodel/lib/active_model/validations/resolve_value.rb

Making a difference between a proc with a parameter or not makes little sense.

Reply:
Ah neat, I didn't realize resolution worked like this. Will fix 👍